tensornetwork/block_sparse/charge.py

```diff
@@ -108,6 +108,17 @@ def charges(self):
     """
     return self.unique_charges[:, self.charge_labels]
 
+  @property
+  def dtype(self):
+    return self.unique_charges.dtype
+
+  @property
+  def degeneracies(self):
+    return np.sum(
```

Comment:
Break this up, (took me too long to figure out what function the ", 0" was supposed to be for).